src/events/api/serializers.py

```diff
@@ -11,117 +11,168 @@ class PrimarySpeakerSerializer(serializers.Serializer):
+
+
 class TalkListSerializer(serializers.ModelSerializer):
     speakers = serializers.SerializerMethodField()
+    is_sponsored = serializers.ReadOnlyField(default=False)
 
     def get_speakers(self, obj):
-        return [
-            ReturnDict(PrimarySpeakerSerializer(
-                data={'thumbnail_url': i.user.get_thumbnail_url(),
-                      'name': i.user.speaker_name,
-                      'github_profile_url': i.user.github_profile_url,
-                      'twitter_profile_url': i.user.twitter_profile_url,
-                      'facebook_profile_url': i.user.facebook_profile_url}).get_initial(),
-                serializer=PrimarySpeakerSerializer) for i in obj.speakers]
+        request = self.context.get('request')
+        users = [s.user for s in obj.speakers]
+        return format_speakers_data(request, users)
 
     class Meta:
         model = TalkProposal
+        fields = ["id", "title", "category", "speakers", "is_sponsored"]
+
+
+class SponsoredEventDetailSerializer(serializers.ModelSerializer):
+    speakers = serializers.SerializerMethodField()
+    is_sponsored = serializers.ReadOnlyField(default=True)
+
+    def get_speakers(self, obj):
+        request = self.context.get('request')
+        return format_speakers_data(request, [obj.host], show_details=True)
+
+    def to_representation(self, obj):
+        """
+        Assign the value of `is_remote` as `SponsoredEvent.remoting_policy`
+        """
+        representation = super().to_representation(obj)
+        is_remote = representation.pop('remoting_policy')
+        representation['is_remote'] = is_remote
+        return representation
+
+    class Meta:
+        model = SponsoredEvent
         fields = [
-            "id",
-            "title",
-            "category",
-            "labels",
-            "speakers",
+            "title", "category", "language", "python_level",
+            "recording_policy", "abstract", "detailed_description",
+            "slide_link", "slido_embed_link", "speakers", "location",
+            "begin_time", "end_time", "remoting_policy",
+            "is_sponsored",
         ]
 
 
-class SponsoredEventSerializer(serializers.ModelSerializer):
+class SponsoredEventListSerializer(serializers.ModelSerializer):
+    speakers = serializers.SerializerMethodField()
+    is_sponsored = serializers.ReadOnlyField(default=True)
 
-    host_name = serializers.CharField(source="host.speaker_name", required=False)
+    def get_speakers(self, obj):
+        request = self.context.get('request')
+        return format_speakers_data(request, [obj.host])
 
     class Meta:
         model = SponsoredEvent
+        fields = ["id", "title", "category", "speakers", "is_sponsored"]
+
+
+class TutorialProposalSerializer(serializers.ModelSerializer):
+    speakers = serializers.SerializerMethodField()
+
+    def get_speakers(self, obj):
+        request = self.context.get('request')
+        users = [s.user for s in obj.speakers]
```

Comment:
I think it is more readable to replace the parameter name from users to speakers.